addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/AlertEntity.java

```diff
@@ -0,0 +1,316 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.oast.internal;
+
+import java.sql.Timestamp;
+import javax.jdo.annotations.Cacheable;
+import javax.jdo.annotations.Column;
+import javax.jdo.annotations.IdGeneratorStrategy;
+import javax.jdo.annotations.Index;
+import javax.jdo.annotations.PersistenceCapable;
+import javax.jdo.annotations.Persistent;
+import javax.jdo.annotations.PrimaryKey;
+import org.datanucleus.api.jdo.annotations.CreateTimestamp;
+import org.parosproxy.paros.core.scanner.Alert;
+
+@Cacheable("false")
+@PersistenceCapable(table = "ALERT", detachable = "true")
+public class AlertEntity {
+
+    @CreateTimestamp private Timestamp createTimestamp;
+
+    @PrimaryKey
+    @Column(length = 512)
+    @Index(name = "ALERT_PAYLOAD_IDX")
+    @Persistent(valueStrategy = IdGeneratorStrategy.IDENTITY)
+    private Integer id;
+
+    private String payload;
+
+    @Column(name = "MESSAGEID")
+    @Persistent(dependent = "true")
+    private MessageEntity message;
+
+    private int alertId;
+    private int pluginId;
+
+    @Column(length = 16777216)
+    private String name;
+
+    private int risk;
+    private int confidence;
+
+    @Column(length = 16777216)
+    private String description;
+
+    @Column(length = 1048576)
+    private String uri;
+
+    @Column(length = 16777216)
+    private String param;
+
+    @Column(length = 32768)
+    private String attack;
+
+    @Column(length = 16777216)
+    private String otherInfo;
+
+    @Column(length = 16777216)
+    private String solution;
+
+    @Column(length = 16777216)
+    private String reference;
+
+    @Column(length = 16777216)
+    private String evidence;
+
+    @Column(length = 256)
+    private String inputVector;
+
+    private int cweId;
+    private int wascId;
+    private int sourceId;
+
+    @Column(length = 256)
+    private String alertRef;
+
+    public AlertEntity(String payload, MessageEntity message, Alert alert) {
+        this.payload = payload;
+        this.message = message;
+
+        alertId = alert.getAlertId();
+        pluginId = alert.getPluginId();
+        name = alert.getName();
+        risk = alert.getRisk();
+        confidence = alert.getConfidence();
+        description = alert.getDescription();
+        uri = alert.getUri();
+        param = alert.getParam();
+        attack = alert.getAttack();
+        otherInfo = alert.getOtherInfo();
+        solution = alert.getSolution();
+        reference = alert.getReference();
+        evidence = alert.getEvidence();
+        inputVector = alert.getInputVector();
+        cweId = alert.getCweId();
+        wascId = alert.getWascId();
+        sourceId = alert.getSource() != null ? alert.getSource().getId() : 0;
```

Comment:
Could you switch this to use positive logic?

I know it'll be the same result but it'll be easier to read for future maintenance

Reply:
What do you mean with positive? `== null`?

Reply:
Changed.

Reply:
Yes, that's what I was after.